### PR TITLE
Add initial search usage doc.

### PR DESCRIPTION
**Pull request by tseaver**
Forked from @jgeewax's version in #972, and updated to match patterns from PubSub/BigQuery/DNS.

### CODE REVIEW

File: docs/search-usage.rst [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+   >>> document = index.document('to-be-deleted')
+   >>> document.exists()  # API call
+   True
+   >>> document.delete()  # API call
+   >>> document.exists()  # API clal
+   False
+
+.. note::
+
+   To update aocuments in place after manipulating its fields or rank, just
```
Comment (dhermes) [inline docs/search-usage.rst]: s/aocuments/documents

File: docs/search-usage.rst [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+   recreate it:  E.g.:
+
+   .. doctest::
+
+      >>> document = index.document('document-id')
+      >>> document.exists()  # API call
+      True
+      >>> document.rank = 12345
+      >>> field = document.field('field-name')
+      >>> field.add_value('christina aguilera')
```
Comment (dhermes) [inline docs/search-usage.rst]: Lulz

File: docs/search-usage.rst [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+
+Fields
+~~~~~~
+
+Fields belong to documents and are the data that actually gets searched.
+
+Each field can have multiple values, which can be of the following types:
+
+- String (Python2 :class:`unicode`, Python3 :class:`str`)
+- Number (Python :class:`int` or :class:`float`)
```
Comment (dhermes) [inline docs/search-usage.rst]: Or class `long` (ducks)

File: docs/search-usage.rst [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+- Timestamp (Python :class:`datetime.datetime`)
+- Geovalue (Python tuple, (:class:`float`, :class:`float`)
```
Comment (dhermes) [inline docs/search-usage.rst]: Paren never gets closed

File: docs/search-usage.rst [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+  tags, and treating the rest of the content like Text.
+
+.. doctest::
+
+   >>> from gcloud import search
+   >>> client = search.Client()
+   >>> index = client.index('index-id')
+   >>> document = index.document('document-id')
+   >>> field = document.field('field-name')
+   >>> field.add_value('britney spearks', tokenization='atom')
```
Comment (dhermes) [inline docs/search-usage.rst]: Also lulz. Did you mean to say "spear**k**s"?

File: docs/search-usage.rst [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+   >>> field.add_value(''<h1>Britney Spears</h1>', tokenization='html')
+
+Searching
+~~~~~~~~~
+
+After populating an indexe with documents, search through them by
```
Comment (dhermes) [inline docs/search-usage.rst]: s/indexe/index

File: docs/search-usage.rst [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+
+.. doctest::
+
+   >>> from gcloud import search
+   >>> client = search.Client()
+   >>> index = client.index('index-id')
+   >>> query = client.query('britney spears')
+   >>> matching_documents = index.search(query)  # API call
+   >>> for document in matching_documents:
+   ...     print document.id
```
Comment (dhermes) [inline docs/search-usage.rst]: Should we attempt to make our doc examples Python3 friendly?

File: docs/search-usage.rst [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+   ['document-id']
+
+By default, all queries are sorted by the ``rank`` value set when the
+document was created.  See:
+https://cloud.google.com/search/reference/rest/v1/projects/indexes/documents#resource_representation.google.cloudsearch.v1.Document.rank
```
Comment (dhermes) [inline docs/search-usage.rst]: This is not a public URI (try it in incognito mode)

/cc @jgeewax

Comment (dhermes) [inline docs/search-usage.rst]: @tseaver Aside about this, if you want to limit to 80 chars, you can just use a slash

```
https://cloud.google.com/search/reference/rest/v1/projects/indexes/documents\
#resource_representation.google.cloudsearch.v1.Document.rank
```

File: docs/search-usage.rst (hunk 5 above)
Comment (tseaver) [inline docs/search-usage.rst]: @jgeewax used BS as his example in #972

File: docs/search-usage.rst (hunk 8 above)
Comment (tseaver) [inline docs/search-usage.rst]: @jgeewax Maybe we sholdn't be implementing a wrapper for an API whose docs are still marked "confidential"?

File: docs/search-usage.rst (hunk 7 above)
Comment (tseaver) [inline docs/search-usage.rst]: e1d9c32

File: docs/search-usage.rst (hunk 6 above)
Comment (tseaver) [inline docs/search-usage.rst]: 6c876df

File: docs/search-usage.rst (hunk 4 above)
Comment (tseaver) [inline docs/search-usage.rst]: b65a370

File: docs/search-usage.rst (hunk 1 above)
Comment (tseaver) [inline docs/search-usage.rst]: 1011298

File: docs/search-usage.rst (hunk 8 above)
Comment (dhermes) [inline docs/search-usage.rst]: :+1:

**Comment by dhermes** [comment]
LGTM

Please hold off on merging until @jgeewax is able to discuss the "confidential" marking part.

File: docs/search-usage.rst (hunk 8 above)
Comment (jgeewax) [inline docs/search-usage.rst]: I already cleared this with the PM for cloud search as part of building it out on gcloud-node. We're all fine to do this -- no worries :)

Comment (dhermes) [inline docs/search-usage.rst]: :+1: 

@tseaver We're all good then (unless you wanted to keep the link under 80 chars)

**Pull request merged**